CIP-0069/README.md

```diff
@@ -0,0 +1,65 @@
+
+## Motivation / History
+
+As it stands the scripts being made on cardano often suffer this problem, and the tokens usually are made to be able to be minted at any time. This leads to further checks being made on the frontend and further fragilitiy of the systems we create. When a mutual dependency arises we are forced to choose which script gets to statically know what's the hash of the other, and which has to be provided 'during runtime'.
+
+Use Case 1: Minting validator checks datum given to spending validator. The spending validator requires the token be present as witness of the datum's correctness.
+
+Use Case 2 (taken from Optim's Liquidity Bonds): Unique NFT is minted to give a unique identifier to a loan, that then gets reused by Bond Tokens. The spending validators require that NFT be present. 
+
+Use Case 3 (taken from Minswap's Dex V1): NFT is minted for the same reason as above. It allows a minting policy to later mint LP tokens with that unique id token name.
+
+We see a similar pattern repeating over and over again as witnessed by dapp developers and auditors alike. By allowing the multi-purpose policies (spending and any other) we increase the security of Cardano by giving us more confidence and allowing to design protocols that have their architecture driven by Cardano's features, not limited by Cardano's language.
+
+This primarily manifests in the ability to use a single validator for both minting and spending but the proposed solution makes it possible to use one validator for any and all purposes at once.
+
+## Specification
+
+### Removing the datum argument
+
+All the script purposes have a form of `Redeemer -> ScriptContext -> exists a. a` except the `Spending` one.
+It has the following form: `Datum -> Redeemer -> ScriptContext -> exists a. a`. This is enforced by the Cardano ledger. 
+
+We modify the general signature to be `ScriptArgs -> ScriptContext -> exists a. a`.
+
+```hs
+data ScriptArgs =
+    RedeemerOnly Redeemer
+  | RedeemerAndDatum Redeemer Datum
+```
+
+## Rationale
+
+Unifying of the script signature is a very elegant solution to the problem, streamlining the experience of developing on cardano.
+Given that accessing the datum is almost always made by a spending script, it makes sense to introduce that argument back to the `ScriptPurpose` that now plays a more important role.
+It begs the question if it should be added as an argument to all validators, to further emphasize that fact. 
+
+## Backwards compatibility
+
+This change is not backwards compatible; it must be introduced in a new Plutus language version.
+Node code must be modified.
+
+## Path to Active
+
+The appropriate changes need to be made to the node code and a new language major version be introduced. 
```

Comment:
This section needs to contain an implementation plan approved by the Plutus team. Once this is covered, we can proceed and merge it as _Proposed_.

Reply:
Let's discuss this in the CIP editors meeting

Reply:
FYI @zygomeb the next meeting, after having been delayed 1 week, has now been postponed 1 additional day ([CIP Editors Meeting #58 (30 Nov, 9:30am UTC)](https://discord.com/events/971785110770831360/1036940834932793363))